### DISCUSSION TITLE
Re-initialize form defaults on submit

### DISCUSSION THREAD

**Discussion by nlrowe** [react-hook-form/react-hook-form#2561] [Q&A]
I have a form that initializes with data using `defaultValues`. The user is allowed to edit the fields and submit the form and I need to reinitialize my `defaultValues`. From the documentation it looks like I should use `reset(data)` to reset the default values, which I have given a go [here](https://codesandbox.io/s/react-hook-form-usefieldarray-nested-arrays-k2nyj?file=/src/index.js).

However, when I click submit; the form fields do not reset to the updated values that I am passing to the `reset` function in the submit handler. They reset to the initial defaults. Anything you see that I can improve?

**Comment by bluebill1049** [comment]
Do you mean this? `reset(defaultValues);`

**Reply by nlrowe** [reply]
Yes. It seems it is not resetting the default values, or I am doing it wrong in the CSB.
